engine/src/main/java/org/terasology/config/PermissionConfig.java

```diff
@@ -13,28 +13,31 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package org.terasology.config;
 
 import java.security.SecureRandom;
 import java.util.ArrayList;
 import java.util.List;
 
-/**
- */
+
 public class PermissionConfig {
     /**
      * Used for first time authentication at a headless server which may be at a remote location.
      */
     private String oneTimeAuthorizationKey = createRandomKey();
 
-    public String getOneTimeAuthorizationKey() {
-        return oneTimeAuthorizationKey;
-    }
-
-    public void setOneTimeAuthorizationKey(String oneTimeAuthorizationKey) {
-        this.oneTimeAuthorizationKey = oneTimeAuthorizationKey;
-    }
-
+    /**
+     * @return Builds a random key on the priniciple stated below:
+     * Create a SecureRandom object, and a list of possible characters
+     * Populate possible characters with A-Z, a-z, and 2-9.
+     * Remove all 'l', 'I' and 'O' characters
+     * Create a StringBuilder called codeBuilder and do this:
+     * Loop 20 times
+     * If 1 is not zero, and you have reached a character that is a multiple of four, add a dash
+     * Dash or not, append something random from possibleCharacters.
+     * Once complete return the StringBuilder codeBuilder.toString();
```

Comment:
@rzats is right. However, before the @return line I would include some details of what is returned, that is: how many characters are returned (20?),  what characters to expect [A-Za-z2-9-] and the fact that 0,O,1,l,I  are not to be expected for security reasons, because they can be confused with each other.